easybuild/easyconfigs/h/hunspell/hunspell-1.7.0-intel-2018b.eb

```diff
@@ -0,0 +1,28 @@
+# Contribution from the Crick HPC team
+# uploaded by J. Sassmannshausen
+
+easyblock = 'ConfigureMake'
+
+name = 'hunspell'
+version = '1.7.0'
+
+homepage = 'http://hunspell.github.io/'
+description = """Hunspell is a spell checker and morphological analyzer library and program designed for languages
+ with rich morphology and complex word compounding or character encoding."""
+
+toolchain = {'name': 'intel', 'version': '2018b'}
```

Comment:
We should use GCCcore for this one surely?

Reply:
I could not get it to work with GCCcore
libtool: link: g++  -fPIC -DPIC -shared -nostdlib /lib/../lib64/crti.o /camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtbeginS.o  .libs/affentry.o .libs/affixmgr.o .libs/csutil.o .libs/filemgr.o .libs/hashmgr.o .libs/hunspell.o .libs/hunzip.o .libs/phonet.o .libs/replist.o .libs/suggestmgr.o   -Wl,-rpath -Wl,/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/../lib64 -Wl,-rpath -Wl,/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/../lib64 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib64 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib -lpthread -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib64/../lib64 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/../lib64 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/libtool/2.4.6-GCCcore-8.2.0/lib -L/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/../../.. /camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/../lib64/libstdc++.so -lm -lc -lgcc_s /camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtendS.o /lib/../lib64/crtn.o  -O2 -march=native   -Wl,-soname -Wl,libhunspell-1.7.so.0 -o .libs/libhunspell-1.7.so.0.0.1
/usr/bin/ld.gold: error: /camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtbeginS.o: unsupported reloc 42 against global symbol _ITM_deregisterTMCloneTable
/usr/bin/ld.gold: error: /camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtbeginS.o: unsupported reloc 42 against global symbol _ITM_registerTMCloneTable
/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtbeginS.o(.text+0x16): error: unsupported reloc 42
/camp/stp/hpc/working/sassy/build/eb-3.8.0/software/GCCcore/8.2.0/lib/gcc/x86_64-pc-linux-gnu/8.2.0/crtbeginS.o(.text+0x57): error: unsupported reloc 42

The original file was with Intel as well so I thought I am at least updating the version to the latest one. I might need to look into the GCCcore one again though.

Reply:
That looks like you missed adding `('binutils', '2.31.1'),` at a build dep.

Reply:
ok, many thanks for that this worked. However, we are puzzled:
$ eb hunspell-1.7.0-GCCcore-8.2.0.eb -D
 * [x] $CFGS/eb-3.8.0/software/EasyBuild/3.8.0/lib/python2.7/site-packages/easybuild_easyconfigs-3.8.0-py2.7.egg/easybuild/easyconfigs/b/binutils/binutils-2.31.1.eb (module: binutils/2.31.1)
So why do I need to add the builddepends here specifically?

I will upload the new build file in a moment and we can discuss that offline.

Reply:
That list shows all _recursive_ build + run dependencies. So, this means that one of your other deps or build-deps needed binutils.
That list isn't loaded when you build a package.

Reply:
Ok. I thought the builddependencies means you only need it for building but not for running. Thanks for the explanation.

Reply:
But that *is* what builddependencies mean; which is why you need to load it explicitly again (because you won't get it indirectly from autotools, as it's just a build-dep there as well).

Reply:
The `binutils/2.31.1` built with the `dummy` toolchain is a *build* dependency for the `GCCcore` toolchain.
To avoid relying on tools built with the `dummy` toolchain as much as possible, we subsequently build a `binutils` with that `GCCcore` toolchain, which is the one you should include as a build dependency whenever building something with `GCCcore`.

Reply:
Thanks all! I will try to remember this (and I am sure I will fail again) :-)